hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALKeyImpl.java

```diff
@@ -573,6 +622,12 @@ public void readFieldsFromPb(WALProtos.WALKey walKey,
     if (walKey.hasOrigSequenceNumber()) {
       this.origLogSeqNum = walKey.getOrigSequenceNumber();
     }
+    if (walKey.getExtendedAttributesCount() > 0){
+      this.extendedAttributes = new HashMap<>(walKey.getExtendedAttributesCount());
+      for (WALProtos.Attribute attr : walKey.getExtendedAttributesList()){
+        extendedAttributes.put(attr.getKey(), attr.getValue().toByteArray());
```

Comment:
Noticed while backporting. Attribute values may have been compressed and need to be decompressed. Will fix up for commit.